src/SDKConnector.php

```diff
@@ -228,46 +235,48 @@ private function setCredentials(CredentialsInterface $credentials) {
    *   The credentials.
    */
   private function buildCredentials(KeyInterface $key): CredentialsInterface {
-    /** @var \Drupal\apigee_edge\Plugin\EdgeKeyTypeInterface $key */
     if ($key->getKeyType() instanceof EdgeKeyTypeInterface) {
       if ($key->getKeyType()->getAuthenticationType($key) === EdgeKeyTypeInterface::EDGE_AUTH_TYPE_OAUTH) {
         return new OauthCredentials($key);
       }
       return new Credentials($key);
     }
-    else {
-      throw new AuthenticationKeyException("Type of {$key->id()} key does not implement EdgeKeyTypeInterface.");
-    }
+
+    throw new AuthenticationKeyException("Type of {$key->id()} key does not implement EdgeKeyTypeInterface.");
   }
 
   /**
    * Generates a custom user agent prefix.
    */
-  protected function userAgentPrefix(): string {
-    if (NULL === self::$userAgentPrefix) {
+  private function userAgentPrefix(): string {
+    if ($this->userAgentPrefix === NULL) {
       $module_info = $this->infoParser->parse($this->moduleHandler->getModule('apigee_edge')->getPathname());
       if (!isset($module_info['version'])) {
         $module_info['version'] = '8.x-1.x-dev';
       }
       // TODO Change "DevPortal" to "Drupal module" later. It has been added for
       // Apigee's convenience this way.
-      self::$userAgentPrefix = $module_info['name'] . ' DevPortal ' . $module_info['version'];
+      $this->userAgentPrefix = $module_info['name'] . ' DevPortal ' . $module_info['version'];
     }
 
-    return self::$userAgentPrefix;
+    return $this->userAgentPrefix;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function testConnection(KeyInterface $key = NULL) {
-    if ($key !== NULL) {
-      $credentials = $this->buildCredentials($key);
-      $client = $this->buildClient($credentials->getAuthentication(), $credentials->getKeyType()->getEndpoint($credentials->getKey()));
+  public function testConnection(KeyInterface $key = NULL): void {
+    // Instead of calling $this->getClient() here we retrieve the current
+    // instance of the SDK connector service and call getClient() on that.
+    // We do this because otherwise SDK connector service decorators that
+    // decorates the getClient() method would not be called.
+    if ($key === NULL) {
+      $client = $this->container->get('apigee_edge.sdk_connector')->getClient();
```

Comment:
Well, this is probably a clear sign of this method should not belong to the SDK connector (as a service and interface) because it depends on the current instance of the SDK connector service.